StripeConnect/StripeConnect/Source/Internal/FinancialConnectionsPresenter.swift

```diff
@@ -0,0 +1,34 @@
+//
+//  FinancialConnectionsPresenter.swift
+//  StripeConnect
+//
+//  Created by Mel Ludowise on 10/18/24.
+//
+
+@_spi(STP) import StripeCore
+import StripeFinancialConnections
+import UIKit
+
+/// Wraps `FinancialConnectionsSheet` for easy dependency injection in tests
+class FinancialConnectionsPresenter {
+    @MainActor
+    func presentForToken(
+        apiClient: STPAPIClient,
+        clientSecret: String,
+        connectedAccountId: String,
+        from presentingViewController: UIViewController
+    ) async -> FinancialConnectionsSheet.TokenResult {
+        let financialConnectionsSheet = FinancialConnectionsSheet(
+            financialConnectionsSessionClientSecret: clientSecret
+        )
+        // FC needs the connected account ID to be configured on the API Client
+        // Make a copy before modifying so we don't unexpectedly modify the shared API client
+        financialConnectionsSheet.apiClient = apiClient.makeCopy()
+        financialConnectionsSheet.apiClient.stripeAccount = connectedAccountId
+        return await withCheckedContinuation { continuation in
```

Comment:
Just wanted to check to see if we have weighed the pros and cons of checked vs unchecked continuation here. If we stick with checked then this __will__ cause a crash if resume is called more than once. It shouldn't get called more than once in this instance, but maybe we can add some extra protections here to ensure we don't call resume more than once and instead of crashing we log an error.

Reply:
I didn't spend too much time thinking about it in this instance since it shouldn't be possible for it to get called more than once. I don't think we have a strong opinion as to which to use the SDK.

Reply:
Ok that's ok, I don't think this should be a blocker or anything. Just wanted to call out that we have been bitten by this before. Especially within an SDK it would be ideal to not crash on a programmer error like this.